scripts/gen_kobject_list.py

```diff
@@ -108,18 +108,94 @@ def write_gperf_table(fp, eh, objs, static_begin, static_end):
     fp.write(footer)
 
 
+driver_macro_tpl = """
+#define _SYSCALL_DRIVER_%(type)s(ptr, op) \\
+	do { \\
+		_SYSCALL_OBJ(ptr, K_OBJ_DRIVER_%(type)s); \\
+		_SYSCALL_DRIVER_OP(ptr, %(api_struct)s, op); \\
+	} while (0)
+"""
```

Comment:
It is easier for users of generated code to understand it when it is human-readable.

Could you look into refactoring away some of the redundancy,
e.g. something like this:

```
#define _SYSCALL_DRIVER_GEN(ptr, op, driver_lower_case, driver_upper_case) \
	do {                                             \
		_SYSCALL_OBJ(ptr, K_OBJ_DRIVER_##driver_upper_case);    \
		_SYSCALL_DRIVER_OP(ptr, driver_lower_case##_driver_api, op); \
	} while (0)

#define _SYSCALL_DRIVER_ADC(ptr, op)     _SYSCALL_DRIVER_GEN(ptr, op, adc, ADC)
#define _SYSCALL_DRIVER_AIO_CMP(ptr, op) _SYSCALL_DRIVER_GEN(ptr, op, aio_cmp, AIO_CMP)


```

Instead of:

```
#define _SYSCALL_DRIVER_COUNTER(ptr, op) \
	do { \
		_SYSCALL_OBJ(ptr, K_OBJ_DRIVER_COUNTER); \
		_SYSCALL_DRIVER_OP(ptr, counter_driver_api, op); \
	} while (0)

#define _SYSCALL_DRIVER_CRYPTO(ptr, op) \
	do { \
		_SYSCALL_OBJ(ptr, K_OBJ_DRIVER_CRYPTO); \
		_SYSCALL_DRIVER_OP(ptr, crypto_driver_api, op); \
	} while (0)


```

Would make it clear at a glance that all _SYSCALL_DRIVER_* calls consist of a _OBJ call and a DRIVER_OP call.

(This also ties into the general principle that you want to do as much as possible in the source language, and as little as possible in the build system / codegen scripts).

Reply:
This is a kind of po-tah-to/po-tay-to situation, but, sure, let me make the change.